src/Microsoft.TestPlatform.Client/DesignMode/DesignModeTestHostLauncher.cs

```diff
@@ -30,15 +31,22 @@ public DesignModeTestHostLauncher(IDesignModeClient designModeClient)
     public virtual bool IsDebug => false;
 
     /// <inheritdoc/>
+    [Obsolete("Don't use, internal implementations should use only the latest interface.", true)]
     public bool AttachDebuggerToProcess(int pid)
     {
-        return _designModeClient.AttachDebuggerToProcess(pid, CancellationToken.None);
+        throw new NotSupportedException();
```

Comment:
We don't catch it explicitly, and this api is never called by us. It throws to make it easier to see when we call the wrong interface version and would lose the additional data. 

It would be better to replace it with internal interface, but it is exposed by IFrameworkHandle.